docs-src/dev-guide/sdk-features.md

```diff
@@ -0,0 +1,571 @@
+
+A Start with Signal won't trigger the Workflow to start immediately and follow the delay.
+
+## Dynamic Entities
+
+Temporal supports Dynamic Workflows, Activities, Signals, and Queries.
+
+These are unnamed entities that are invoked if no other statically defined entity with the given name exists.
+
+| Feature                                   | Go | Java | TypeScript | Python | .Net | PHP |
+| ----------------------------------------- | -- | ---- | ---------- | ------ | ---- | --- |
+| [Dynamic Activities](#dynamic-activities) | No | Yes  | No         | Yes    | Yes  | No  |
+| [Dynamic Queries](#dynamic-queries)       | No | Yes  | No         | Yes    | Yes  | No  |
+| [Dynamic Signals](#dynamic-signals)       | No | Yes  | Yes        | Yes    | Yes  | No  |
+| [Dynamic Workflows](#dynamic-workflows)   | No | Yes  | Yes        | Yes    | Yes  | No  |
+
+Dynamic Entities should be used judiciously as a fallback mechanism rather than the primary approach.
+Overusing them can lead to maintainability and debugging issues down the line.
+
+## Enhanced stack trace
+
+| Feature                                       | Go | Java | TypeScript | Python | .Net | PHP |
+| --------------------------------------------- | -- | ---- | ---------- | ------ | ---- | --- |
+| [Enhanced stack trace](#enhanced-stack-trace) | No | No   | Yes        | No     | No   | No  |
+
+Enhanced stack trace in Temporal is a feature that allows you to capture and view detailed information about the execution of a Workflow.
+It provides a stack trace of all the threads owned by the Workflow Execution, which can be useful for troubleshooting issues in production.
+The `__stack_trace` query is a predefined query available in many SDKs that returns the stack trace.
+You can use the `tctl Workflow stack` command to query the stack trace of a Workflow Execution.
+This feature is especially helpful for identifying errors and blocks in a Workflow Definition.
+
+## gRPC interceptors
+
+| Feature                                 | Go  | Java | TypeScript | Python | .Net | PHP |
+| --------------------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [gRPC interceptors](#grpc-interceptors) | Yes | Yes  | Yes        | No     | No   | No  |
+
+gRPC interceptors serve as a powerful tool provided by gRPC, enabling developers to intercept and modify both incoming RPC requests and outgoing RPC responses.
+This mechanism can be instrumental in scenarios such as dynamic adjustments to target hosts or tweaking parameters like TLS configurations.
+However, for those utilizing the Temporal SDKs powered by the Rust core, which includes the TypeScript, Python, and .Net SDKs.
+
+Currently, features like gRPC interceptors aren't implemented directly in the Rust core.
+This limitation extends to other scenarios like the rotation of client TLS certificates.
+Although technically feasible, the current workaround involves recreating the client to reflect changes.
+
+## Health service
```

Comment:
Health service is a cluster feature not a SDK feature. The SDK feature may be to expose a call to invoke it, but this description is a description for the cluster part. There are dozens of other client calls more important than this one.